geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/registry/ProtobufOperationContextRegistry.java

```diff
@@ -45,6 +46,13 @@ public OperationContext getOperationContext(RequestAPICase apiCase) {
   }
 
   private void addContexts() {
+    operationContexts.put(RequestAPICase.AUTHENTICATIONREQUEST,
+        new OperationContext<>(ClientProtocol.Request::getAuthenticationRequest,
+            new AuthenticationRequestOperationHandler(),
+            opsResp -> ClientProtocol.Response.newBuilder().setAuthenticationResponse(opsResp),
+            new ResourcePermission(ResourcePermission.Resource.DATA,
```

Comment:
I realize that this should never be used, but is there a more generic permission we can specify than DATA, READ (maybe something like ANY,ANY).

Reply:
Not sure what we can do here... but we never authorize authentication messages. The handler should not allow authentication to happen more than once

Reply:
I think we allow authentication messages to pass when they are authorized to read data... but we will fail with nice message when they try and process the message